metaflow/plugins/argo/argo_workflows_decorator.py

```diff
@@ -23,10 +31,47 @@ def task_pre_step(
         inputs,
     ):
         self.task_id = task_id
+        self.run_id = run_id
+
+        triggers = []
+        # Expose event triggering metadata through current singleton
+        for key, payload in os.environ.items():
+            if key.startswith("METAFLOW_ARGO_EVENT_PAYLOAD_"):
+                if payload != "null":  # Argo-Workflow's None
+                    try:
+                        payload = json.loads(payload)
+                    except (TypeError, ValueError) as e:
+                        # There could be arbitrary events that Metaflow doesn't know of
```

Comment:
But this payload definitely came in through `METAFLOW_ARGO_EVENT_PAYLOAD_` right?  I wonder if we really should be swallowing this.  I think it's more likely that this is a bug in the setup (either in code or in supporting infra).

The end result is downstream flow gets run, but without the right payload - confusing to debug (first hand experience!).

Instead this COULD just throw an error indicating that the event payload was invalid (and possibly print it).

Reply:
It may come in through `METAFLOW_ARGO_EVENT_PAYLOAD_ ` but might still result from an event we don't control (hence outside the purview of the event triggering). If the event is a valid JSON then this will work as is.